server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/change/StateChange.scala

```diff
@@ -31,7 +31,7 @@ trait TypeName {
 
   def asString(): String
   def parse(string: String): Option[TypeName]
-  def parseState(string: String): Either[IllegalArgumentException, State]
+  def parseState(string: String): Either[IllegalArgumentException, State] = UuidState.parse(string)
```

Comment:
No I think this change is a regression. 

Less line of code is not always good: It is perfectly fine to have all factories use 1 line to specify how they deserialise stuff explicitly...